
# Table of Contents
1. [Introduction](#introduction)
2. [Project Description](#project-description)
3. [Tools](#tools)
4. [Project Log](#project-log)
5. [Contact](#contact)

## Introduction
Is bioinformatics or computational biology more awesome? We do not know. However, this image illustrates what the field is all about!
![](logo.jpg)
In other words bioinformatics is insane... Mindblown!

[About]()
[Members]()
[Project Log]()

Also here is a link to a page [page2](page_2)
## Project description
This project is the final part of the  __Applied Bioinformatics__ course that is part of the [Swedish graduate school of medical bioinformatics](http://www.medbioinfo.se/). The course ran 10-14 June 2019.

The aim of the project is to use good coding practices and version control in order to ensure reproducibility and readability of results and data.

The bioninformatical task is to trim multiple-sequence alignment (MSA) based on [entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)). As homologous proteins can contain regions not inherited or regions with too many changes to properly align. These regions will have a higher variation that the rest of the alignment and will result in noisy phylogeny estimates. Because of the expected higher variation of positions in these regions the entropy will also increase making entropy a viable filtering criteria.

Given a simulated dataset with a known true phylogeny we can estimate the distance from a trimmed phylogeny estimate to the true phylogeny using [Robinson-Foulds distance](https://en.wikipedia.org/wiki/Robinson-Foulds_metric).

We also compare our results to widely used trimming software [TrimAl](http://trimal.cgenomics.org/).

## Tools
## Project Log
## Contact

